### PR TITLE
NEWS: add v3.1.0, v3.0.16 + minor fixes

### PR DESCRIPTION
**Pull request by gyuho**

### CODE REVIEW

File: NEWS [hunk 1]
```diff
@@ -1,3 +1,25 @@
...
+- discovery now has upper limit for waiting on retries
+- get responses are sorted in ascending order by default
+- add etcd --log-output flag
+- add etcd --metrics flag
+- add lease time-to-live API; fetch lease information
+- add v3 etcdctl get --print-value-only flag
+- add v3 etcdctl make-mirror --dest-prefix flag
+- add clientv3 Sync, SetEndpoints; update endpoints at runtime
+- drop clientv3.Config.Logger field
+- recipies now embed concurrency.Session
```
Comment (xiang90) [inline NEWS]: i do not understand thisone

File: NEWS [hunk 2]
```diff
@@ -1,3 +1,25 @@
...
+  (i.e., example.com) if no custom certificate authority is given
+- discovery now has upper limit for waiting on retries
+- get responses are sorted in ascending order by default
+- add etcd --log-output flag
+- add etcd --metrics flag
+- add lease time-to-live API; fetch lease information
+- add v3 etcdctl get --print-value-only flag
+- add v3 etcdctl make-mirror --dest-prefix flag
+- add clientv3 Sync, SetEndpoints; update endpoints at runtime
+- drop clientv3.Config.Logger field
```
Comment (xiang90) [inline NEWS]: it is replaced by the global logger setup? shall we mention it here?

File: NEWS [hunk 3]
```diff
@@ -1,3 +1,25 @@
...
+- SRV records (e.g., infra1.example.com) must match the discovery domain
+  (i.e., example.com) if no custom certificate authority is given
+- discovery now has upper limit for waiting on retries
+- get responses are sorted in ascending order by default
+- add etcd --log-output flag
+- add etcd --metrics flag
+- add lease time-to-live API; fetch lease information
+- add v3 etcdctl get --print-value-only flag
+- add v3 etcdctl make-mirror --dest-prefix flag
+- add clientv3 Sync, SetEndpoints; update endpoints at runtime
```
Comment (xiang90) [inline NEWS]: separate this into two lines?

File: NEWS (hunk 1 above)
Comment (gyuho) [inline NEWS]: @xiang90 It was for https://github.com/coreos/etcd/pull/6172.

Should we mention that? It's API change in `contrib/recipes`.
Maybe just `remove session manager and add ttl in recipes`?

Comment (xiang90) [inline NEWS]: @heyitsanthony Can you try to rephrase this? Thanks!

Comment (heyitsanthony) [inline NEWS]: `- recipes now conform to sessions defined in clientv3/concurrency`?

File: NEWS [hunk 4]
```diff
@@ -1,3 +1,26 @@
...
+- etcd --strict-reconfig-check flag is set by default
+- cluster rejects removing members if quorum will be lost
+- SRV records (e.g., infra1.example.com) must match the discovery domain
+  (i.e., example.com) if no custom certificate authority is given
+- discovery now has upper limit for waiting on retries
+- get responses are sorted in ascending order by default
+- add etcd --log-output flag
+- add etcd --metrics flag
+- add lease time-to-live API; fetch lease information
+- add v3 etcdctl get --print-value-only flag
```
Comment (xiang90) [inline NEWS]: probably make the etcdctl, clientv3 changes a separated section?

etcdctl
 - xxx

clientv3
 - yyy

?

**Comment by xiang90** [comment]
lgtm. defer to @heyitsanthony

**Comment by gyuho** [comment]
I added a line about https://github.com/coreos/etcd/pull/7145.

File: NEWS [hunk 5]
```diff
@@ -1,3 +1,32 @@
+etcd v3.1.0 (2017-01-13)
+- faster linearizable reads (implements Raft read-index)
+- automatic leadership transfer when leader steps down
+- etcd uses default host IP if advertise URL is not given
+- cluster rejects removing members if quorum will be lost
+- SRV records (e.g., infra1.example.com) must match the discovery domain
+  (i.e., example.com) if no custom certificate authority is given
+  - TLSConfig ServerName is ignored if custom certificate is given
+- discovery now has upper limit for waiting on retries
+- etcd flag
```
Comment (heyitsanthony) [inline NEWS]: `etcd flags`

File: NEWS [hunk 6]
```diff
@@ -1,3 +1,32 @@
+etcd v3.1.0 (2017-01-13)
+- faster linearizable reads (implements Raft read-index)
+- automatic leadership transfer when leader steps down
+- etcd uses default host IP if advertise URL is not given
```
Comment (heyitsanthony) [inline NEWS]: s/host/route

File: NEWS [hunk 7]
```diff
@@ -1,3 +1,32 @@
...
+- cluster rejects removing members if quorum will be lost
+- SRV records (e.g., infra1.example.com) must match the discovery domain
+  (i.e., example.com) if no custom certificate authority is given
+  - TLSConfig ServerName is ignored if custom certificate is given
```
Comment (heyitsanthony) [inline NEWS]: TLSConfig ServerName is ignored with user-provided certificates for backwards compatibility; to be deprecated in 3.2

File: NEWS [hunk 8]
```diff
@@ -1,3 +1,32 @@
...
+  - replace Config.Logger field with global logger
+  - Get API responses are sorted in ascending order by default
+- v3 etcdctl
+  - add lease timetolive command
+  - add --print-value-only flag to get command
+  - add --dest-prefix flag to make-mirror command
+  - command get responses are sorted in ascending order by default
+- recipes now conform to sessions defined in clientv3/concurrency
+- ACI has symlinks to /usr/local/bin/etcd*
+
```
Comment (heyitsanthony) [inline NEWS]: - warn on binding listeners through domain names; to be deprecated in 3.2

**Comment by gyuho** [comment]
fyi. just added notes about  v3 auth is stable, and experimental grpc proxy

**Comment by heyitsanthony** [comment]
lgtm

**Pull request merged**